pkg/controller/storagecluster/controller_test.go

```diff
@@ -6238,6 +6238,145 @@ func TestNodeShouldRunStoragePod(t *testing.T) {
 	require.True(t, shouldContinueRunning)
 }
 
+func TestDoesTelemetryMatch(t *testing.T) {
```

Comment:
Instead of testing a private method, we could add a test like this - https://github.com/libopenstorage/operator/blob/master/pkg/controller/storagecluster/controller_test.go#L3322
This tests the complete reconcile flow and validates that the pod is marked for deletion when change is detected.

Reply:
Kept existing for loop as wanted a UT that targets that function directly. Easier to focus incremental changes.

Also added a Reconcile test now for the same cases.